### PR TITLE
Drop support for Unity 4, to support push notifications.

### LINKED ISSUE #36: Unity 5.2.1f iOS Push notification Exception
Hi i'm integrating parse in my Unity Project, Object creation works fine but I'm having some trouble with the push notification, the device never seem to register with the server on iOS.
using 5.2.1p3

Using ilcpp backend, Link.xml is located in the Asset's folder root. 

<pre>
ExecutionEngineException: Attempting to call method 'System.Reflection.MonoProperty::StaticGetterAdapterFrame' for which no ahead of time (AOT) code was generated.
  at System.Reflection.MonoProperty+GetterAdapter.Invoke (System.Object _this) [0x00000] in <filename unknown>:0 
  at System.Reflection.MonoProperty.GetValue (System.Object obj, System.Object[] index) [0x00000] in <filename unknown>:0 
  at Parse.PlatformHooks+<>c__DisplayClass2c.<RegisteriOSPushNotificationListener>b__2b () [0x00000] in <filename unknown>:0 
  at System.Action.Invoke () [0x00000] in <filename unknown>:0 
  at Parse.PlatformHooks+<RunDispatcher>d__2e.MoveNext () [0x00000] in <filename unknown>:0 
UnityEngine.Debug:Internal_LogException(Exception, Object)
UnityEngine.Debug:LogException(Exception)
Parse.<RunDispatcher>d__2e:MoveNext()
</pre>

### PR DESCRIPTION
**Pull request by richardjrossiii**
Push notifications on Unity 5 were broken due to IL2CPP not supporting PropertyInfo being exported by any type.

Fixes #36.

### CODE REVIEW

File: Parse/PlatformHooks.Unity.cs [hunk 1]
```diff
@@ -1023,36 +1023,15 @@ private class SettingsWrapper : IDictionary<string, object> {
     /// </summary>
     /// <param name="action">Action to be completed when device token is received.</param>
     internal static void RegisterDeviceTokenRequest(Action<byte[]> action) {
-      // Basically what we want to achieve here is:
```
Comment (hallucinogen) [inline Parse/PlatformHooks.Unity.cs]: I'm glad I left this comment

**Comment by hallucinogen** [comment]
This looks good to me. Let's punt this to `1.6.2` or `1.7.0` since it's a breaking change.

**Comment by richardjrossiii** [comment]
Hmm. Seeing as these methods are currently broken in the 1.6 release, do we really want to keep them broken (for both unity 4 and 5) in the next release?

**Comment by hallucinogen** [comment]
You're correct. Let's land this then

**Comment by hallucinogen** [comment]
Merge this maybe?

**Comment by richardjrossiii** [comment]
My bad, PR wasn't assigned to me so I didn't see it :)

**Pull request merged**